### ISSUE TITLE
wilor into smplx

### ISSUE THREAD

**Issue by AreinDaralnakhla** [rolpotamias/WiLoR#33]
How to align wilor's 3D keypoints with SMPLX hand keypoints?

**Comment by rolpotamias** [comment]
Can you elaborate a bit? Do you have 2D/3D landmarks from both MANO/SMPL-X? or you want to transfere the hand poses of WiLoR to SMPLX?

**Comment by rolpotamias** [comment]
You can modify the [demo.py ](https://github.com/rolpotamias/WiLoR/blob/main/demo.py#L98) to flip the left/right hand accordingly to match the SMPLX format. 
A quick script to do that (might need some adaptation): 

        pose_axis_angle = matrix_to_axis_angle(out['pred_mano_params']['hand_pose']).reshape(-1, 15*3)
        global_axis_angle = matrix_to_axis_angle(out['pred_mano_params']['global_orient']).reshape(-1, 3).detach().clone().cpu()
        poses_final = torch.zeros_like(pose_axis_angle.detach().clone().cpu())
        for i in range(len(all_right)): 
            poses_final[i] = pose_axis_angle[i].detach().clone().cpu()
            
            if all_right[i] == 0: 
                mean_hand = hands_mean_left
            else:
                mean_hand = hands_mean_right
            
            if all_right[i] == 0:
                poses_final[i] *= -1 
                poses_final[i][::3] *= -1 

            poses_final[i]  -= mean_hand

where `hands_mean_right, hands_mean_left` are from the .pkl files downloaded from [here](https://mano.is.tue.mpg.de/):

`hands_mean_right = torch.from_numpy(pd.read_pickle('/PATH_TO_/MANO_RIGHT.pkl')['hands_mean']).float()
       hands_mean_left  = torch.from_numpy(pd.read_pickle('/PATH_TO_//MANO_LEFT.pkl')['hands_mean']).float()
`

for the `matrix_to_axis_angle` transform you can refer to this [gist](https://gist.github.com/rolpotamias/0d51b0b6ee0710986a5dec45cf28c0b2)

**Comment by AreinDaralnakhla** [comment]
but doesnt flat_hand_mean=True inside smplx layer handle that already?

**Comment by AreinDaralnakhla** [comment]
the above didnt fix the issue for why passing the wilor left hand poses (after necessary conversion and miroring) in smplx layer gives worse result than the right hand
